smarts/core/tests/test_mutiple_traffic_data.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+import shutil
+import tempfile
+
+from pathlib import Path
+from unittest.mock import MagicMock
+from smarts.core.scenario import Scenario
+from smarts.core.traffic_history_provider import TrafficHistoryProvider
+from smarts.sstudio.sumo2mesh import generate_glb_from_sumo_network
+from smarts.sstudio import gen_social_agent_missions, gen_missions
+from smarts.sstudio.types import Mission, Route, SocialAgentActor
+
+AGENT_ID = "Agent-007"
+
+
+@pytest.fixture
+def scenario_parent_path():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        yield temp_dir
+
+
+@pytest.fixture
+def create_scenario(scenario_parent_path):
+    scenario = Path(scenario_parent_path) / "cycles"
+    scenario.mkdir()
+
+    shutil.copyfile(
+        Path(__file__).parent / "maps/6lane.net.xml", scenario / "map.net.xml"
+    )
+    generate_glb_from_sumo_network(
+        str(scenario / "map.net.xml"), str(scenario / "map.glb")
+    )
```

Comment:
There is a utility in `smarts.core.tests.helpers` that does this. You can look at: https://github.com/huawei-noah/SMARTS/blob/99b689a7e01d891242b13b32de7f871753c7b8df/smarts/core/tests/test_boids.py#L32-L52 

`yield` in pytest fixtures means the remaining after the yield will be run when the test ends for purposes of cleanup.

Reply:
Thanks! I initially copied the above code from `smarts/core/tests/test_scenario.py` . Should I apply the same there?

Also, I noticed `make test` is not picking up `test_scenario.py` as one of the test files, and `pytest smarts/core/tests/test_scenario.py` also doesn't run the test, which seems strange

Reply:
> Thanks! I initially copied the above code from `smarts/core/tests/test_scenario.py` . Should I apply the same there?  

Yes, if that is not a problem.
> Also, I noticed `make test` is not picking up `test_scenario.py` as one of the test files, and `pytest smarts/core/tests/test_scenario.py` also doesn't run the test, which seems strange  

Thanks for noticing could you change the following line to `def test_scenario_variations_of_social_agents(scenario_root):`

https://github.com/huawei-noah/SMARTS/blob/99b689a7e01d891242b13b32de7f871753c7b8df/smarts/core/tests/test_scenario.py#L70